app/src/main/java/com/github/se/project/ui/overview/Overview.kt

```diff
@@ -276,35 +276,27 @@ private fun TutorSections(
     onClick: (Lesson) -> Unit,
     listProfilesViewModel: ListProfilesViewModel
 ) {
-  val sections =
-      if (lessons.any { it.status == LessonStatus.INSTANT_CONFIRMED }) {
-        listOf(
-            SectionInfo(
-                "Instant Lesson",
-                LessonStatus.INSTANT_CONFIRMED,
-                ImageVector.vectorResource(
-                    id = R.drawable.bolt_24dp_000000_fill1_wght400_grad0_opsz24)),
-            SectionInfo(
-                "Waiting for your Confirmation",
-                LessonStatus.PENDING_TUTOR_CONFIRMATION,
-                Icons.Default.Notifications),
-            SectionInfo(
-                "Waiting for the Student Confirmation",
-                LessonStatus.STUDENT_REQUESTED,
-                ImageVector.vectorResource(id = R.drawable.baseline_access_time_24)),
-            SectionInfo("Upcoming Lessons", LessonStatus.CONFIRMED, Icons.Default.Check))
-      } else {
-        listOf(
-            SectionInfo(
-                "Waiting for your Confirmation",
-                LessonStatus.PENDING_TUTOR_CONFIRMATION,
-                Icons.Default.Notifications),
-            SectionInfo(
-                "Waiting for the Student Confirmation",
-                LessonStatus.STUDENT_REQUESTED,
-                ImageVector.vectorResource(id = R.drawable.baseline_access_time_24)),
-            SectionInfo("Upcoming Lessons", LessonStatus.CONFIRMED, Icons.Default.Check))
-      }
+  val sections = mutableListOf<SectionInfo>()
+
+  if (lessons.any { it.status == LessonStatus.INSTANT_CONFIRMED }) {
+    sections.add(
+        SectionInfo(
+            "Upcoming Instant Lesson",
+            LessonStatus.INSTANT_CONFIRMED,
+            ImageVector.vectorResource(
+                id = R.drawable.bolt_24dp_000000_fill1_wght400_grad0_opsz24)))
+  }
+  sections.add(
+      SectionInfo(
+          "Waiting for your Confirmation",
+          LessonStatus.PENDING_TUTOR_CONFIRMATION,
+          Icons.Default.Notifications))
+  sections.add(
```

Comment:
I am not too sure about that implementation and I would like to have Alexis opinion on it @AlexisAllmnd